### PR TITLE
PCHR-1251: My leave report html

### PR DESCRIPTION
**Pull request by dhavalsharma**
The html for my leave page as per PCHR-1251. 

The page will look like this 
my-leave-report

A detailed description of breaking down this page into various components is described in [PR 117](https://github.com/compucorp/civihr-employee-portal-theme/pull/117)

### CODE REVIEW

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 1]
```diff
@@ -25,10 +25,155 @@
             <div class="tab-content">
                 <div id="report" class="tab-pane fade in active">
                     <h3>Report</h3>
-                    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit,
-                        sed do eiusmod tempor incididunt ut labore et dolore
-                        magna aliqua.
-                    </p>
+
+                    <div class="container-fluid civi-container">
+                        <div class="civi-container__title">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Indentation should be 2 spaces (not tabs)

File: civihr_employee_portal/css/custom.css [hunk 2]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+body{
+    font-family: 'Lato', sans-serif;
+}
+
+.civi-container{
+    /*width: 80%;*/
+    margin-top: 60px;
+    background-color: rgb(247, 247, 249);
+    z-index: 46;
+
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Unnecessary new lines

File: civihr_employee_portal/css/custom.css [hunk 3]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+/* styles for my leave report */
+body{
+    font-family: 'Lato', sans-serif;
+}
+
+.civi-container{
+    /*width: 80%;*/
+    margin-top: 60px;
+    background-color: rgb(247, 247, 249);
+    z-index: 46;
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Magic number, why `46`?

File: civihr_employee_portal/css/custom.css [hunk 4]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
   vertical-align: middle;
   padding-left: 10px;
 }
+
+/* styles for my leave report */
+body{
+    font-family: 'Lato', sans-serif;
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Why are you defining a `body` rule here in the middle of this file? Isn't the `font-family` defined somewhere else already?

File: civihr_employee_portal/css/custom.css [hunk 5]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+}
+
+.civi-container{
+    /*width: 80%;*/
+    margin-top: 60px;
+    background-color: rgb(247, 247, 249);
+    z-index: 46;
+
+
+}
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: You must leave a new line between one selector and the other
```css
.selector-1 {

}

.selector-2 {

}

File: civihr_employee_portal/css/custom.css [hunk 6]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+.civi-separator {
+    background-color: rgb(237, 237, 237);
+    height: 2px;
+    z-index: 55;
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Another magic number

File: civihr_employee_portal/css/custom.css [hunk 7]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+.civi-container{
+    /*width: 80%;*/
+    margin-top: 60px;
+    background-color: rgb(247, 247, 249);
+    z-index: 46;
+
+
+}
+.civi-separator {
+    background-color: rgb(237, 237, 237);
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Why are you using the rgb notation? Why not the hex notation?

File: civihr_employee_portal/css/custom.css [hunk 8]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+    height: 2px;
+    z-index: 55;
+}
+
+.civi-container__title{
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Always a space between the selector and the `{`
```css
.civi-container__title {
```

File: civihr_employee_portal/css/custom.css [hunk 9]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+
+}
+.civi-grid--closed{
+
+}
+
+.civi-grid__header{
+    background-color: rgb(255, 255, 255);
+    box-shadow: 0px 6px 20px 0px rgba(0, 0, 0, 0.03),0px 8px 17px 0px rgba(0, 0, 0, 0.03);
+    z-index: 45;
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Magic number

File: civihr_employee_portal/css/custom.css [hunk 10]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+}
+.civi-grid__body{
+    margin: 32px;
+    background-color: rgb(255, 255, 255);
+    box-shadow: 0px 6px 20px 0px rgba(0, 0, 0, 0.03),0px 8px 17px 0px rgba(0, 0, 0, 0.03);
+    z-index: 74;
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Magic number

File: civihr_employee_portal/css/custom.css [hunk 11]
```diff
@@ -159,3 +159,72 @@ span.appraisals-employee-legend {
...
+
+.civi-grid__header{
+    background-color: rgb(255, 255, 255);
+    box-shadow: 0px 6px 20px 0px rgba(0, 0, 0, 0.03),0px 8px 17px 0px rgba(0, 0, 0, 0.03);
+    z-index: 45;
+}
+.civi-grid__body{
+    margin: 32px;
+    background-color: rgb(255, 255, 255);
+    box-shadow: 0px 6px 20px 0px rgba(0, 0, 0, 0.03),0px 8px 17px 0px rgba(0, 0, 0, 0.03);
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: This `box-shadow` is the same as the one above, maybe it's a generic `box-shadow` value that you can put in a sass variable?

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 12]
```diff
@@ -24,11 +24,255 @@
...
                 <div id="report" class="tab-pane fade in active">
-                    <h3>Report</h3>
-                    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit,
-                        sed do eiusmod tempor incididunt ut labore et dolore
-                        magna aliqua.
-                    </p>
+
+                    <div class="chr-container">
+                      <div class="row chr-container__row">
+                          <div class="col-md-9 ">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: You are using a 4 spaces indentation. You should use 2 spaces

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 13]
```diff
@@ -24,11 +24,255 @@
...
+                                                      <tr class="civi-grid--open ">
+                                                          <th colspan="2"><i class="fa fa-chevron-down" aria-hidden="true"></i> Period Entitlement</th>
+                                                          <th colspan="2"></th>
+                                                          <th colspan=""></th>
+                                                          <th>31</th>
+                                                          <th>3</th>
+                                                          <th>0</th>
+                                                          <th>0</th>
+                                                          <th>0</th>
+
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Please make sure you are not leaving around empty lines

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 14]
```diff
@@ -24,11 +24,255 @@
...
+                                                          <td></td>
+                                                          <td></td>
+                                                          <td></td>
+                                                          <td></td>
+                                                          <td><i class="fa fa-ellipsis-v" aria-hidden="true"></i></td>
+                                                      </tr>
+                                                      </tbody>
+
+                                                  </table>
+                                              </div><!--end of .table-responsive-->
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: There is no need for this kind of comments, please remove them

File: civihr_employee_portal/css/custom.css [hunk 15]
```diff
@@ -159,3 +159,5 @@ span.appraisals-employee-legend {
   vertical-align: middle;
   padding-left: 10px;
 }
+
+/* styles for my leave report */
```
Comment (AkA84) [inline civihr_employee_portal/css/custom.css]: Why you kept this?

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 16]
```diff
@@ -24,11 +24,255 @@
...
+                              </div>
+                          </div>
+                      </div><!-- panel group -->
+
+                      <div class="row chr-container__row chr-container__row--summary ">
+                          <!---<div class="col-xs-12">-->
+                              <div class="table-responsive">
+                                  <table class="table chr-table__borderless">
+                                      <thead>
+                                      <tr class="">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Why an empty `[class]`?

Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: You are not indenting

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 17]
```diff
@@ -24,11 +24,255 @@
...
+                                          <th class="" colspan="6">
+                                            <p >Future Balance</p>
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Make sure your syntax is correct

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 18]
```diff
@@ -24,11 +24,255 @@
...
+                                                          <th></th>
+                                                      </tr>
+                                                      </thead>
+                                                  </table>
+                                              </div>
+                                          </div>
+                                      </a>
+                                  </h4>
+                              </div>
+                              <div id="collapseOne" class="panel-collapse collapse in">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: why are you using camelCase for the `[id]`?

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 19]
```diff
@@ -24,11 +24,255 @@
...
-                        sed do eiusmod tempor incididunt ut labore et dolore
-                        magna aliqua.
-                    </p>
+
+                    <div class="chr-container">
+                      <div class="row chr-container__row">
+                          <div class="col-md-9 ">
+                            <p>
+                              <a href="#" class=""><i class="fa fa-download" aria-hidden="true"></i>Export CSV</a>
+                            </p>
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Why are you wrapping the link in a paragraph?

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 20]
```diff
@@ -24,11 +24,255 @@
...
-                    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit,
-                        sed do eiusmod tempor incididunt ut labore et dolore
-                        magna aliqua.
-                    </p>
+
+                    <div class="chr-container">
+                      <div class="row chr-container__row">
+                          <div class="col-md-9 ">
+                            <p>
+                              <a href="#" class=""><i class="fa fa-download" aria-hidden="true"></i>Export CSV</a>
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Why the `[aria-hidden]` param?

**Comment by AkA84** [comment]
@dhavalsharma the PR is completely voided of any description, screenshots or code snippets. There are strict guidelines on how a PR should be presented in order to be approved, please follow them

**Comment by AkA84** [comment]
> PCHR-1251: added rest of HTML

As stated in the guidelines, the commit title should be in the imperative mood (http://chris.beams.io/posts/git-commit/)

> PCHR-1251: Add rest of HTML

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 21]
```diff
@@ -11,29 +11,574 @@
...
-            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
-          </p>
-        </div>
-        <div id="calendar" class="tab-pane fade">
-          <h3>Calendar</h3>
-          <p>
-            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodoconsequat.
-          </p>
+    <div class="tabs-container">
+        <div class="container">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: You are still using a 4 spaces indentation

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 22]
```diff
@@ -11,29 +11,574 @@
...
+                <li role="presentation" class="active">
+                    <a href="#report">Report</a>
+                </li>
+                <li role="presentation">
+                    <a href="#calendar">Calendar</a>
+                </li>
+            </ul>
+            <div class="tab-content">
+                <div id="report" class="tab-pane fade in active">
+
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Empty line, as said previously, make sure you clean your code. You can't give a PR for review if you haven't first made sure that the code is *final*

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 23]
```diff
@@ -11,29 +11,574 @@
...
+                    <div class="container-fluid chr-container">
+                      <div class="row chr-container__row">
+                          <div class="col-md-9 ">
+                              <a href="#"><i class="fa fa-download" aria-hidden="true"></i><span class="chr-margin-left-10">Export CSV</span></a>
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Please break lines that are too long
```html
<a href="#">
  <i class="fa fa-download" aria-hidden="true"></i>
  <span class="chr-margin-left-10">Export CSV</span>
</a>
```
also, why do you need the `[aria-hidden=true]`?  The icon is not hidden

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 24]
```diff
@@ -11,29 +11,574 @@
...
+                          </div>
+                          <div class="col-md-3">
+                              <!-- Split button -->
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: As said before, you don't need this kind of comments

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 25]
```diff
@@ -11,29 +11,574 @@
...
+                                      <li><a href="#">Action</a></li>
+                                      <li><a href="#">Another action</a></li>
+                                      <li><a href="#">Something else here</a></li>
+                                      <li role="separator" class="divider"></li>
+                                      <li><a href="#">Separated link</a></li>
+                                  </ul>
+                              </div>
+                          </div>
+                      </div>
+
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Empty lines (same below)

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 26]
```diff
@@ -11,29 +11,574 @@
...
+                                                      <th>0</th>
+                                                  </tr>
+                                                </thead>
+                                            </table>
+                                        </div>
+                                    </div>
+
+                                </h4>
+                            </div>
+                            <div id="collapseOne" class="panel-collapse collapse in">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: again, you are using camelCase for the `[id]`, but you should use the dasherized-case

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 27]
```diff
@@ -11,29 +11,574 @@
...
+                                          <th>-1</th>
+                                          <th></th>
+                                      </tr>
+                                      </thead>
+                                  </table>
+                              </div>
+                          <!--</div>-->
+                      </div>
+
+                        <div class="row panel panel-default chr-grid">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Wrong indentation

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html (hunk 23 above)
Comment (dhavalsharma) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: The icon is still visible to the user but is not read to inaccessible users (with disability), more [here](http://stackoverflow.com/questions/31107040/whats-the-difference-between-html-hidden-and-aria-hidden-attributes)

**Comment by AkA84** [comment]
* The paddings in the headings do not match the design, they're way bigger than they're supposed to be
* the paddings of the balance/future balance sections are way bigger than they're supposed to be
* The heading of the collapsible sections ("Period Entitlement", "Public Holidays", etc) are missing the left padding, the caret is too close to the edge of the element
* The table columns for the leave types (Vacation, Maternity, etc) seem larger compared to the design

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 28]
```diff
@@ -24,26 +24,573 @@
...
+                              <td class="chr-grid__row__cell--light"></td>
+                              <td class="chr-grid__row__cell--light">1</td>
+                              <td class="chr-grid__row__cell--light"></td>
+                              <td class="chr-grid__row__cell--light"></td>
+                              <td class="chr-grid__row__cell--light"></td>
+                              <td><i class="fa fa-ellipsis-v" aria-hidden="true"></i></td>
+                            </tr>
+                          </tbody>
+                        </table>
+                      </div><!--end of .table-responsive-->
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: You need to get rid of these comments

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 29]
```diff
@@ -24,26 +24,573 @@
...
+            <div class="row panel panel-default chr-grid">
+              <div class="panel-heading chr-grid__panel-heading">
+                <h4 class="panel-title">
+                    <div class="row ">
+                      <div class="table-responsive">
+                        <table class="table table-borderless">
+                          <thead>
+                            <tr class="civi-grid--open ">
+                              <th colspan="2">
+                                <i class="fa fa-chevron-down" aria-hidden="true"></i>
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: You still have these `[aria-hidden="true"]` attributes. I keep pointing them out asking why you have them, but you neither remove them or you reply saying why you use them.

Either you explain why or you need to remove them

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 30]
```diff
@@ -24,26 +24,573 @@
...
+                </div><!--panel body -->
+              </div>
+            </div>
+            <div class="row chr-container__row chr-container__row--summary ">
+            <div class="table-responsive">
+              <table class="table table-borderless">
+                <thead>
+                  <tr>
+                    <th colspan="6">
+                    <p >Future Balance</p>
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Malformed tag

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 31]
```diff
@@ -24,26 +24,573 @@
...
+              </div>
+            </div>
+            <div class="row panel panel-default chr-grid">
+              <div class="panel-heading chr-grid__panel-heading">
+                <h4 class="panel-title">
+                    <div class="row ">
+                      <div class="table-responsive">
+                        <table class="table table-borderless">
+                          <thead>
+                          <tr class="civi-grid--open ">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Wrong indentation
```html
<thead>
  <tr>
  <!-- ... -->
</thead>

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 32]
```diff
@@ -24,26 +24,573 @@
...
+            <!-- public holidays end -->
+            <div class="row panel panel-default chr-grid">
+              <div class="panel-heading chr-grid__panel-heading">
+                <h4 class="panel-title">
+                    <div class="row ">
+                      <div class="table-responsive">
+                        <table class="table table-borderless">
+                          <thead>
+                            <tr class="civi-grid--open ">
+                              <th colspan="2"><i class="fa fa-chevron-down" aria-hidden="true"></i>
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]:
```html
<th colspan="2">
  <i class="fa fa-chevron-down" aria-hidden="true"></i>
  Approved Requests
</th>

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 33]
```diff
@@ -24,26 +24,573 @@
...
+                    </div>
+                  </div>
+                </div><!--panel body -->
+              </div>
+            </div>
+            <!-- public holidays start -->
+            <div class="row panel panel-default chr-grid">
+              <div class="panel-heading chr-grid__panel-heading">
+                <h4 class="panel-title">
+                    <div class="row ">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Wrong indentation
```html
<h4 class="panel-title">
<div class="row ">
```

Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Also, malformed attribute value
```html
<div class="row">
```

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 34]
```diff
@@ -24,26 +24,573 @@
...
+                    <li><a href="#">Another action</a></li>
+                    <li><a href="#">Something else here</a></li>
+                    <li role="separator" class="divider"></li>
+                    <li><a href="#">Separated link</a></li>
+                  </ul>
+                </div>
+              </div>
+            </div>
+            <div class="chr-separator"></div>
+            <div class="row  chr-container__row">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Two spaces between class names
```html
<div class="row chr-container__row">
```

File: civihr_employee_portal/civihr_leave_absences/templates/my_leave.html [hunk 35]
```diff
@@ -24,26 +24,573 @@
       </ul>
       <div class="tab-content">
         <div id="report" class="tab-pane fade in active">
-          <h3>Report</h3>
-          <p>
-            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
-          </p>
+          <div class="container-fluid chr-container">
+            <div class="row chr-container__row">
+              <div class="col-md-9 ">
```
Comment (AkA84) [inline civihr_employee_portal/civihr_leave_absences/templates/my_leave.html]: Malformed class value
```html
<div class="col-md-9">
```

**Pull request closed without merging**